balancer-js/src/modules/zaps/bbausd2-migrations/bbausd1.ts

```diff
@@ -0,0 +1,312 @@
+      .div(totalLiquidity)
+      .toString();
+    const usdtBptAmt = BigNumber.from(bbausd1Amount)
+      .sub(usdcBptAmt)
+      .sub(daiBptAmt)
+      .toString();
+
+    const swaps: BatchSwapStep[] = [
+      {
+        poolId: this.addresses.bbausd1.id,
+        assetInIndex: 10,
+        assetOutIndex: 2,
+        amount: daiBptAmt,
+        userData: '0x',
+      },
+      {
+        poolId: this.addresses.linearDai1.id,
+        assetInIndex: 2,
+        assetOutIndex: 1,
+        amount: '0',
+        userData: '0x',
+      },
+      {
+        poolId: this.addresses.linearDai2.id,
+        assetInIndex: 1,
+        assetOutIndex: 3,
+        amount: '0',
+        userData: '0x',
+      },
+      {
+        poolId: this.addresses.bbausd2.id,
+        assetInIndex: 3,
+        assetOutIndex: 0,
+        amount: '0',
+        userData: '0x',
+      },
+      {
+        poolId: this.addresses.bbausd1.id,
+        assetInIndex: 10,
+        assetOutIndex: 5,
+        amount: usdcBptAmt,
+        userData: '0x',
+      },
+      {
+        poolId: this.addresses.linearUsdc1.id,
+        assetInIndex: 5,
+        assetOutIndex: 4,
+        amount: '0',
+        userData: '0x',
+      },
+      {
+        poolId: this.addresses.linearUsdc2.id,
+        assetInIndex: 4,
+        assetOutIndex: 6,
+        amount: '0',
+        userData: '0x',
+      },
+      {
+        poolId: this.addresses.bbausd2.id,
+        assetInIndex: 6,
+        assetOutIndex: 0,
+        amount: '0',
+        userData: '0x',
+      },
+      {
+        poolId: this.addresses.bbausd1.id,
+        assetInIndex: 10,
+        assetOutIndex: 8,
+        amount: usdtBptAmt,
+        userData: '0x',
+      },
+      {
+        poolId: this.addresses.linearUsdt1.id,
+        assetInIndex: 8,
+        assetOutIndex: 7,
+        amount: '0',
+        userData: '0x',
+      },
+      {
+        poolId: this.addresses.linearUsdt2.id,
+        assetInIndex: 7,
+        assetOutIndex: 9,
+        amount: '0',
+        userData: '0x',
+      },
+      {
+        poolId: this.addresses.bbausd2.id,
+        assetInIndex: 9,
+        assetOutIndex: 0,
+        amount: '0',
+        userData: '0x',
+      },
+    ];
+
+    // For tokens going in to the Vault, the limit shall be a positive number. For tokens going out of the Vault, the limit shall be a negative number.
+    const limits = [
+      BigNumber.from(minBbausd2Out).mul(-1).toString(), // bbausd2
+      '0',
+      '0',
+      '0',
+      '0',
+      '0',
+      '0',
+      '0',
+      '0',
+      '0',
+      bbausd1Amount, // Max in should be bbausd1 amount
+    ];
+
+    // Swap to/from Relayer
+    const funds: FundManagement = {
+      sender,
+      recipient,
+      fromInternalBalance: false,
+      toInternalBalance: false,
+    };
+
+    const encodedBatchSwap = Relayer.encodeBatchSwap({
+      swapType: SwapType.SwapExactIn,
+      swaps,
+      assets,
+      funds,
+      limits,
+      deadline: MaxUint256,
```

Comment:
we might want to set it to shorter time, like 1h, just in case we end up with some crazy bug or edge case

Reply:
✅